Mathlib/Algebra/Lie/Nilpotent.lean

```diff
@@ -205,14 +215,30 @@ instance (priority := 100) trivialIsNilpotent [IsTrivial L M] : IsNilpotent R L
   ⟨by use 1; change ⁅⊤, ⊤⁆ = ⊥; simp⟩
 #align lie_module.trivial_is_nilpotent LieModule.trivialIsNilpotent
 
-theorem nilpotent_endo_of_nilpotent_module [hM : IsNilpotent R L M] :
+theorem exists_forall_pow_toEndomorphism_eq_zero [hM : IsNilpotent R L M] :
     ∃ k : ℕ, ∀ x : L, toEndomorphism R L M x ^ k = 0 := by
   obtain ⟨k, hM⟩ := hM
   use k
   intro x; ext m
   rw [LinearMap.pow_apply, LinearMap.zero_apply, ← @LieSubmodule.mem_bot R L M, ← hM]
   exact iterate_toEndomorphism_mem_lowerCentralSeries R L M x m k
-#align lie_module.nilpotent_endo_of_nilpotent_module LieModule.nilpotent_endo_of_nilpotent_module
+#align lie_module.nilpotent_endo_of_nilpotent_module LieModule.exists_forall_pow_toEndomorphism_eq_zero
+
+theorem isNilpotent_toEndomorphism_of_isNilpotent [IsNilpotent R L M] (x : L) :
+    _root_.IsNilpotent (toEndomorphism R L M x) := by
+  change ∃ k, toEndomorphism R L M x ^ k = 0
+  have := exists_forall_pow_toEndomorphism_eq_zero R L M
+  tauto
+
+theorem isNilpotent_toEndomorphism_of_isNilpotent₂ [IsNilpotent R L M] (x y : L) :
+    _root_.IsNilpotent (toEndomorphism R L M x ∘ₗ toEndomorphism R L M y) := by
+  obtain ⟨k, hM⟩ := (inferInstance : IsNilpotent R L M)
```

Comment:
Is 
```
  obtain ⟨k, hM⟩ : IsNilpotent R L M := inferInstance
```
better or worse? Is it more readable? Does the fact that you're doing this at all mean that there's some missing lemma? The thing is, you'd have to give the type anyway and
```
  obtain ⟨k, hM⟩ : IsNilpotent R L M := isNilpotent_spec
```
is even longer.

Reply:
You're right: there's a gap in the API here. I'll plug it.